plugins/optimization-detective/class-od-link-collection.php

```diff
@@ -248,9 +248,33 @@ public function get_response_header(): ?string {
 		$link_headers = array();
 
 		foreach ( $this->get_prepared_links() as $link ) {
-			// The about:blank is present since a Link without a reference-uri is invalid so any imagesrcset would otherwise not get downloaded.
-			$link['href'] = isset( $link['href'] ) ? esc_url_raw( $link['href'] ) : 'about:blank';
-			$link_header  = '<' . $link['href'] . '>';
+			// Encode only the filename part of the URL to ensure it contains only ASCII characters.
+			if ( isset( $link['href'] ) ) {
+				$parsed_url = wp_parse_url( $link['href'] );
+				if ( isset( $parsed_url['path'] ) ) {
+					$path_segments        = explode( '/', $parsed_url['path'] );
+					$last_segment         = array_pop( $path_segments );
+					$encoded_last_segment = rawurlencode( $last_segment );
```

Comment:
Should this conditionally only encode the `$last_segment` if it contains any characters which are not ASCII?